spec/models/vm/retirement_management_spec.rb

```diff
@@ -170,24 +162,19 @@
 
   it "#raise_retirement_event without current user" do
     event_name = 'foo'
-    event_hash = {:vm => @vm, :host => @vm.host, :type => "ManageIQ::Providers::Vmware::InfraManager::Vm",
-                  :retirement_initiator => "system"}
-    options = {:zone => @vm.my_zone}
+    event_hash = {:userid => nil, :vm => @vm, :host => @vm.host, :type => "ManageIQ::Providers::Vmware::InfraManager::Vm"}
 
-    expect(MiqEvent).to receive(:raise_evm_event).with(@vm, event_name, event_hash, options).once
+    expect(MiqEvent).to receive(:raise_evm_event).with(@vm, event_name, event_hash, nil).once
 
     @vm.raise_retirement_event(event_name)
   end
 
   it "#raise_retirement_event with current user" do
-    user = FactoryGirl.create(:user_with_group, :userid => 'freddy')
     event_name = 'foo'
-    event_hash = {:vm => @vm, :host => @vm.host, :type => "ManageIQ::Providers::Vmware::InfraManager::Vm",
-                  :retirement_initiator => "user", :userid => 'freddy'}
-    options = {:zone => @vm.my_zone}
+    event_hash = {:userid => nil, :vm => @vm, :host => @vm.host, :type => "ManageIQ::Providers::Vmware::InfraManager::Vm"}
 
     User.with_user(user) do
-      expect(MiqEvent).to receive(:raise_evm_event).with(@vm, event_name, event_hash, options).once
+      expect(MiqEvent).to receive(:raise_evm_event).with(@vm, event_name, event_hash, nil).once
```

Comment:
Since you have a `current_user` (via `User.with_user(user)`), I'm expecting a user to flow through here (specifically `user`)